dashboard/utils.py

```diff
@@ -501,3 +501,13 @@ async def _looper(*args, **kwargs):
         return _looper
 
     return _wrapper
+
+
+async def wait_for_stub(stubs: dict, timeout_s: int = 10):
+    i = 0
+    while not stubs:
+        if i == timeout_s:
+            return False
+        i += 1
+        await asyncio.sleep(1)
```

Comment:
use an actual time comparison instead of iteration count btw? this could be wildly off if there is a lot of contention among coroutines

Reply:
removed as well